### PR TITLE
Fix #2828 : Removed unused parameter from AppVersionViewModel.kt

### LINKED ISSUE #2828: Remove unused paramater in AppVersionViewModel
In `AppVersionViewModel` `fragment` is unused. Remove it.

### PR DESCRIPTION
**Pull request by YashKr01**
## Explanation

Fixes #2828 :  removed unused parameter form AppVersionViewModel.kt

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: 2828".)
- [x] The PR explanation includes the words "Fixes #bugnum: 2828" (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Review by MaskedCarrot — CHANGES_REQUESTED** [review]
@YashKr01 revert changes in `.idea/misc.xml`

**Comment by YashKr01** [comment]
@MaskedCarrot okay sir

**Comment by YashKr01** [comment]
@MaskedCarrot sir should i close this current PR and create another after reverting changes?

**Comment by MaskedCarrot** [comment]
@YashKr01 you can change that in the current PR. and also there is no need to call me sir :-)

**Comment by MaskedCarrot** [comment]
you can use `git revert <commit>`

**Comment by MaskedCarrot** [comment]
@YashKr01  before your pr can be reviewed can you please complete the following tasks
1. change the PR title to `Fix #2828: Removed unused parameter from AppVersionViewModel.kt`
2. add the line `Fixes #2828` under the heading `Explanation` in the pr description.
3. Update the checkbox in the PR description to [x] (which means the box is checked)
4. revert the changes made to `.idea/misc.xml`
5. remove the unused import from `AppVersionViewModel.kt` you can do that by using `ctrl+alt+o` in `AppVersionViewModel.kt`

**Comment by YashKr01** [comment]
@MaskedCarrot didn't get it, how to revert changes in current PR?

**Comment by MaskedCarrot** [comment]
@YashKr01  use `git revert b60ce48` in the terminal. you can read more about this [here](https://stackoverflow.com/a/21718540/12505541)

File: .idea/misc.xml [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
 <project version="4">
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="true" project-jdk-name="1.8" project-jdk-type="JavaSDK">
     <output url="file://$PROJECT_DIR$/build/classes" />
   </component>
   <component name="ProjectType">
     <option name="id" value="Android" />
   </component>
-</project>
+</project>
```
Comment (rt4914) [inline .idea/misc.xml]: Please revert changes from this file.

File: app/src/main/java/org/oppia/android/app/administratorcontrols/appversion/AppVersionViewModel.kt [hunk 2]
```diff
@@ -15,7 +15,6 @@ import javax.inject.Inject
 /** [ViewModel] for [AppVersionFragment]*/
 @FragmentScope
 class AppVersionViewModel @Inject constructor(
-  fragment: Fragment,
```
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/administratorcontrols/appversion/AppVersionViewModel.kt:18]: You should remove the unused `import` related to this too.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@YashKr01 Please add some description to your PR.

In description make sure you mention `Fix #2828` For more details check the `checklist` present in PR description.

File: app/src/main/java/org/oppia/android/app/administratorcontrols/appversion/AppVersionViewModel.kt (hunk 2 above)
Comment (YashKr01) [reply app/src/main/java/org/oppia/android/app/administratorcontrols/appversion/AppVersionViewModel.kt:18]: @rt4914 i've added description on the PR.
And after removing import and reverting the file how should i push changes in this PR ?

**Comment by anandwana001** [comment]
@YashKr01 Please complete all the installation steps 
https://github.com/oppia/oppia-android/wiki#installation

**Comment by YashKr01** [comment]
@MaskedCarrot @rt4914 please have a look now. I've reverted .idea/misc.xml and removed unused import in AppVersionViewModel.kt .

**Review by FareesHussain — CHANGES_REQUESTED** [review]
@YashKr01 .idea/misc.xml is not yet reverted you can see the changes [here](https://github.com/oppia/oppia-android/pull/2835/files).

I suggest you change this manually from github (to avoid the auto-generated code before commit).

Edit [this](https://github.com/YashKr01/oppia-android/blob/remove-unused-parameter/.idea/misc.xml) file and paste the contents of https://github.com/oppia/oppia-android/blob/develop/.idea/misc.xml

Also update the PR description 

`Fix #2835` -> `Fixes #2828`

and PR title 

`FIX #2828` -> `Fix #2828`

**Comment by rt4914** [comment]
> @YashKr01 .idea/misc.xml is not yet reverted you can see the changes [here](https://github.com/oppia/oppia-android/pull/2835/files).
> 
> I suggest you change this manually from github (to avoid the auto-generated code before commit).
> 
> Edit [this](https://github.com/YashKr01/oppia-android/blob/remove-unused-parameter/.idea/misc.xml) file and paste the contents of https://github.com/oppia/oppia-android/blob/develop/.idea/misc.xml
> 
> Also update the PR description
> 
> `Fix #2835` -> `Fixes #2828`
> 
> and PR title
> 
> `FIX #2828` -> `Fix #2828`

@FareesHussain Direct changes on Github is not a good Practice. 

@YashKr01 Suggest doing following:
Copy contents from https://github.com/oppia/oppia-android/blob/develop/.idea/misc.xml and paste in your branch's `misc.xml` file and then make a commit.

File: app/src/main/java/org/oppia/android/app/administratorcontrols/appversion/AppVersionViewModel.kt (hunk 2 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/android/app/administratorcontrols/appversion/AppVersionViewModel.kt:18]: @YashKr01 I think you have already done that.

**Comment by FareesHussain** [comment]
> @FareesHussain Direct changes on Github is not a good Practice.

Yes, but I mentioned it cause android studio generates the same changes even after reverting. but I guess that was only in the case of .idea/gradle.xml

**Comment by YashKr01** [comment]
@FareesHussain @rt4914 as directed i've changed misc.xml and changed PR's title and description

**Review by FareesHussain — APPROVED** [review]
LGTM, thanks

**Review by MaskedCarrot — APPROVED** [review]
LGTM! Thanks.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**